message_edit: Fix copy paste bug on local-echo/pending message.

tushar912 opened the pull request:
Fixes #18053

**Testing plan:** <!-- How have you tested? -->
Tested manually for all cases.

timabbott commented:
@tushar912 can you explain a bit more what happens when we change the `data-clipboard-target` and what the other changes to how we use the ClipboardJS API here do?  Excerpts from the ClipboardJS documentation would be helpful.

Also, it looks like we now have somewhat complicated code copied 3 times; seems like a good time to extract a function.

Signior-X commented:
@timabbott  The copy button isn't working because the clipbaord.js is unable to get the selector of the element.
By the logic we use in `local_messages.js`, the failed messages are locally echoed with a float id containing decimal, and clipboard.js uses `document.querySelector` that doesn't work with an id if it contains a period(.).
This PR fixes this by not using the selector used by clipboard.js and passing the trigger to the element manually escaping it. Mentioned [here in docs](https://clipboardjs.com/#advanced-usage)

> Also, it looks like we now have somewhat complicated code copied 3 times; seems like a good time to extract a function.

I also feel making a function above will be helpful. @tushar912

tushar912 commented:
@Signior-X Thanks for the clarification! I will make a new function for that code.

tushar912 commented:
@timabbott @amanagr After reading the [documentation](https://clipboardjs.com/#advanced-usage) again I realised that we don't even need the ```data-clipboard-target``` attribute. Instead we can dynamically set the target using ```document.querySelector```. So I removed that attribute. I have also extracted the function.

timabbott commented:
I merged this after squashing and renaming the function to be more descriptive.

We should be sure to remember needing to update this if/when we add a "Copy to clipboard" button even with the FULL editability type.

tushar912 commented:
@timabbott I would keep this in mind next time.
